auth/standard_role_manager.cc

```diff
@@ -248,9 +248,13 @@ future<> standard_role_manager::start() {
             }
             return make_ready_future<>();
         }).then([this] {
-            _stopped = auth::do_after_system_ready(_as, [this] {
-                return seastar::async([this] {
-                    if (legacy_mode(_qp)) {
+            auto work_done = seastar::make_lw_shared<seastar::promise<>>();
+            auto ret = work_done->get_future();
+            _stopped = auth::do_after_system_ready(_as, [this, work_done] mutable {
```

Comment:
I think the code can be simplified by assigning the return value of do_after_system_ready() to a shared_future, which can then be copies to _stopped and ret.

But, do we even need _stopped? If start() waits for do_after_system_ready, why do we need stopped()? There's no work to wait for.